### PR TITLE
HttpExecutor addings

### PR DESCRIPTION
**Pull request by wmedvede**
- bring back classes to the corresponding directory
    - improvements on the produced request to consider the HttpRecipientPayload type

### CODE REVIEW

**Comment by wmedvede** [comment]
Hi @tiagodolphine would you mind take a look.

File: jobs-service/jobs-recipients/job-http-recipient/runtime/src/main/java/org/kie/kogito/job/http/recipient/HttpJobExecutor.java [hunk 1]
```diff
@@ -68,11 +71,24 @@ private Uni<HttpResponse<Buffer>> executeCallback(HTTPRequestCallback request) {
                 uri.getPort(),
                 uri.getHost(),
                 uri.getPath()).timeout(timeout);
-        clientRequest.queryParams().addAll(request.getQueryParams());
-        clientRequest.headers().addAll(request.getHeaders());
-        return request.getBody()
-                .map(clientRequest::sendJson)
-                .orElseGet(clientRequest::send);
+        addNonNullEntries(clientRequest.queryParams(), request.getQueryParams());
```
Comment (tiagodolphine) [inline jobs-service/jobs-recipients/job-http-recipient/runtime/src/main/java/org/kie/kogito/job/http/recipient/HttpJobExecutor.java]: I'd say the previous approach is cleaner and easier to read, but anyway up to you

File: jobs-service/jobs-recipients/job-http-recipient/runtime/src/main/java/org/kie/kogito/job/http/recipient/HttpJobExecutor.java [hunk 2]
```diff
@@ -68,11 +71,24 @@ private Uni<HttpResponse<Buffer>> executeCallback(HTTPRequestCallback request) {
...
+    }
+
+    private Buffer buildBuffer(Object body) {
+        if (body instanceof HttpRecipientBinaryPayloadData) {
+            return Buffer.buffer(((HttpRecipientBinaryPayloadData) body).getData());
+        } else if (body instanceof HttpRecipientStringPayloadData) {
+            return Buffer.buffer(((HttpRecipientStringPayloadData) body).getData());
+        }
+        return null;
     }
```
Comment (tiagodolphine) [inline jobs-service/jobs-recipients/job-http-recipient/runtime/src/main/java/org/kie/kogito/job/http/recipient/HttpJobExecutor.java]: I don't get why you need all these `instance of` here... I mean in the end the same getData is called, if we need to generate a buffer from each of the type maybe we can add this into the payload itself... a chain of instance of in general is not a good smell, what happens if  we introduce a new payload in the API? you need to change this block of code in the job-service...

File: jobs-service/jobs-recipients/job-http-recipient/runtime/src/main/java/org/kie/kogito/job/http/recipient/HttpJobExecutor.java [hunk 3]
```diff
@@ -151,4 +167,13 @@ private int getRepeatableJobCountDown(JobDetails job) {
     public Class<HttpRecipient> type() {
         return HttpRecipient.class;
     }
+
+    private static void addNonNullEntries(MultiMap target, Map<String, String> source) {
+        if (source != null) {
```
Comment (tiagodolphine) [inline jobs-service/jobs-recipients/job-http-recipient/runtime/src/main/java/org/kie/kogito/job/http/recipient/HttpJobExecutor.java]: this is a case where the functional style shows its value, and avoids the method to change the input arguments instances and instead returns the value to be assigned to whoever called.

```
request.getHeaders()
                .entrySet()
                .stream()
                .filter(entry -> StringUtils.isNotEmpty(entry.getValue()))
                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```
`request.getHeaders().putAll( filtered); 
.....
`

Comment (wmedvede) [reply jobs-service/jobs-recipients/job-http-recipient/runtime/src/main/java/org/kie/kogito/job/http/recipient/HttpJobExecutor.java]: No sure what you mean, at this point we don't know if the request.getHeaders() is null or not

**Comment by wmedvede** [comment]
@tiagodolphine I think this is ready to go, comments addressed

**Review by tiagodolphine — APPROVED** [review]

**Pull request merged**